csv-validator-ui/src/main/scala/uk/gov/nationalarchives/csv/validator/ui/CsvValidatorUi.scala

```diff
@@ -366,35 +368,43 @@ object CsvValidatorUi extends SimpleSwingApplication {
       }
 
 
-    btnValidate.reactions += onClick(displayWait(
-      suspendUi = {
-        btnValidate.enabled = false
-        this.peer.setCursor(Cursor.getPredefinedCursor(Cursor.WAIT_CURSOR))
-        this.progressBar.value = 0
-        this.progressBar.visible = true
-
-      },
-      action = {
-        txtArReport.text = ""
-        CsvValidatorUi.this.validate(
-          txtCsvFile.getText,
-          settingsPanel.csvEncoding,
-          txtCsvSchemaFile.getText,
-          settingsPanel.csvSchemaEncoding,
-          settingsPanel.failOnFirstError,
-          settingsPanel.pathSubstitutions,
-          settingsPanel.enforceCaseSensitivePathChecks,
-          Some(progress),
-          settingsPanel.validateUtf8
-        )
-      },
-      output = outputToReport,
-      resumeUi = {
-        btnValidate.enabled = true
-        this.peer.setCursor(Cursor.getDefaultCursor)
-        //this.progressBar.visible = false
-      }
-    ))
+    btnValidate.reactions += validateOnClick(btnValidate, false)
+    btnValidateMetadataOnly.reactions += validateOnClick(btnValidate, true)
```

Comment:
I think you need to pass the btnValidateMetadataOnly in this reaction, otherwise it might enable / disable wrong button. Also, if the processing is taking long, I am wondering whether we should disable both buttons?

Reply:
Ah good spot, thanks.